12-offer-encoding.md

```diff
@@ -282,6 +373,40 @@ A writer of an offer:
       - MUST set `offer_quantity_max` to 0.
   - otherwise:
     - MUST NOT set `offer_quantity_max`.
+  - If an offer MAY trigger time-spaced invoice requests:
+    - MUST include exactly one of `offer_recurrence_optional` or `offer_recurrence_compulsory`.
+    - MAY use `offer_recurrence_optional` if a payment of a single period would be useful (compatibility with pre-recurrence readers).
+    - MUST NOT use `offer_recurrence_optional` if `offer_recurrence_base` is present.
+  - if it includes either `offer_recurrence_optional` or `offer_recurrence_compulsory`:
+      - MUST set `time_unit` to 0 (seconds), 1 (days), 2 (months) or 3 (years).
+      - MUST set `period` to how often (in `time-unit`) it wants to be paid.
```

Comment:
Should we add an explicit constraint for `period != 0` to avoid ambiguity? Something like:

Writer → MUST NOT set `period` to 0
Reader → MUST NOT respond to offers with `period` set to 0